app/windows/winget/winget-create.js

```diff
@@ -0,0 +1,153 @@
+  process.exit(1);
+}
+
+let sha256;
+
+const arch = {
+  x64: 'x64',
+  x86: 'x86',
+};
+
+async function createYAMLFiles(version, downloadURL, date, notesURL) {
+  const templates = [
+    'Headlamp.Headlamp.installer.yaml',
+    'Headlamp.Headlamp.locale.en-US.yaml',
+    'Headlamp.Headlamp.yaml',
+  ];
+
+  // use path to get the template directory
+  const templateDir = path.join(__dirname, 'templates/');
+
+  const manifestDir = path.join(directoryPath, '/', version, '/');
+
+  try {
+    // create manifest version directory if it doesnt exist
+    if (!fs.existsSync(manifestDir)) {
+      console.log(`Created directory ${manifestDir}`);
+      fs.mkdirSync(manifestDir, { recursive: true });
+    }
+  } catch (err) {
+    console.error(`Error creating directory ${manifestDir}: ${err}`);
+  }
+
+  templates.forEach(template => {
+    // read the template file
+    const templateContent = fs.readFileSync(`${templateDir}${template}`, 'utf8');
+
+    // replace the template variables with the actual values
+    const newContent = templateContent
+      .replace(/__PACKAGE_VERSION__/g, version)
+      .replace('__DISPLAY_VERSION__', version)
+      .replace('__INSTALLER_SHA256__', sha256)
+      .replace('__INSTALLER_URL__', downloadURL)
+      .replace('__RELEASE_DATE__', date)
+      .replace('__RELEASE_NOTES_URL__', notesURL);
+
+    // write the new content to the manifest directory
+    fs.writeFileSync(`${manifestDir}${template}`, newContent);
+    console.log(
+      `Created winget manifest file ${template} for version ${version} in ${manifestDir}`
+    );
+  });
+}
+
+// Async function to fetch release information from GitHub API for a given tag
+async function fetchGithubReleaseInfo() {
+  const tag = projectVersion;
+  const res = await fetch(
+    `https://api.github.com/repos/headlamp-k8s/headlamp/releases/tags/v${tag}`
+  );
+
+  if (!res.ok) {
+    console.error(
+      `Error fetching release information for tag ${tag}: ${res.status} ${res.statusText}`
+    );
+    process.exit(1);
+  }
+
+  const resJSON = await res.json();
+
+  await updateReleaseInfo(resJSON, tag);
+}
+
+async function getChecksums(checksumsUrl) {
+  const res = await fetch(checksumsUrl);
+  const checksums = await res.text();
+  const checksumMap = {};
+
+  for (const line of checksums.split('\n')) {
+    const [checksum, filename] = line.split('  ');
+
+    if (!filename) {
+      continue;
+    }
+
+    if (filename.includes(arch.x64)) {
+      checksumMap[arch.x64] = checksum;
+    }
+  }
+
+  return checksumMap;
+}
+
+async function updateReleaseInfo(response, tag) {
+  const assets = response.assets;
+  const checksums = assets.find(v => v.name === 'checksums.txt');
+  const checksumsUrl = checksums.browser_download_url;
+  const checksumMap = await getChecksums(checksumsUrl);
+  const browserDownloadUrlInfo = assets.find(
+    asset => asset.name === `Headlamp-${tag}-win-x64.exe`
```

Comment:
This generated file name can be used for a `getChecksum(checksumsURL, fileName)` function.